client/python/openlineage/client/transport/http.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from urllib.parse import urljoin
+
+import attr
+
+from typing import Optional
+
+from requests import Session
+from requests.adapters import HTTPAdapter
+from urllib3.util import parse_url
+
+from openlineage.client.run import RunEvent
+from openlineage.client.serde import Serde
+from openlineage.client.transport.transport import Config, Transport
+from openlineage.client.utils import get_only_specified_fields
+
+log = logging.getLogger(__name__)
+
+
+@attr.s
+class HttpConfig(Config):
+    url: str = attr.ib()
+    timeout: float = attr.ib(default=5.0)
+    # check TLS certificates
+    verify: bool = attr.ib(default=True)
+    api_key: Optional[str] = attr.ib(default=None)
```

Comment:
Rather than an api key directly, let's add a `TokenProvider` type. This enables us to add some oauth or other token generator easily in the future.

Reply:
Duplicate comment